src/number_theory/von_mangoldt.lean

```diff
@@ -0,0 +1,121 @@
+
+/-!
+# The von Mangoldt Function
+
+In this file we define the von Mangoldt function: the function on natural numbers that returns
+`log p` if the input can be expressed as `p^k` for a prime `p`.
+
+## Main Results
+
+The main definition for this file is
+
+- `nat.arithmetic_function.von_mangoldt`: The von Mangoldt function `Λ`.
+
+We then prove the classical summation property of the von Mangoldt function in
+`nat.arithmetic_function.von_mangoldt_sum`, that `∑ i in n.divisors, Λ i = real.log n`, and use this
+to deduce alternative expressions for the von Mangoldt function via Möbius inversion, see
+`nat.arithmetic_function.sum_moebius_mul_log_eq`.
+
+## Notation
+
+We use the standard notation `Λ` to represent the von Mangoldt function.
+
+-/
+
+namespace nat
+namespace arithmetic_function
+
+open finset
+
+open_locale arithmetic_function
+
+/-- `log` as an arithmetic function `ℕ → ℝ`. Note this is in the `nat.arithmetic_function`
+namespace to indicate that it is bundled as an `arithmetic_function` rather than being the usual
+real logarithm. -/
+noncomputable def log : arithmetic_function ℝ :=
+⟨λ n, real.log n, by simp⟩
+
+@[simp] lemma log_apply {n : ℕ} : log n = real.log n := rfl
+
+/-- In the case when `n` is a prime power, `min_fac` will give the appropriate prime. -/
+noncomputable def von_mangoldt : arithmetic_function ℝ :=
+⟨λ n, if is_prime_pow n then real.log (min_fac n) else 0, if_neg not_is_prime_pow_zero⟩
+
+localized "notation `Λ` := nat.arithmetic_function.von_mangoldt" in arithmetic_function
+
+lemma von_mangoldt_apply {n : ℕ} :
+  Λ n = if is_prime_pow n then real.log (min_fac n) else 0 := rfl
+
+@[simp] lemma von_mangoldt_apply_one : Λ 1 = 0 := by simp [von_mangoldt_apply]
+
+@[simp] lemma von_mangoldt_nonneg {n : ℕ} : 0 ≤ Λ n :=
+begin
+  rw [von_mangoldt_apply],
+  split_ifs,
+  { exact real.log_nonneg (one_le_cast.2 (nat.min_fac_pos n)) },
+  refl
+end
+
+lemma von_mangoldt_apply_pow {n k : ℕ} (hk : k ≠ 0) : Λ (n ^ k) = Λ n :=
+by simp only [von_mangoldt_apply, is_prime_pow_pow_iff hk, pow_min_fac hk]
+
+lemma von_mangoldt_apply_prime {p : ℕ} (hp : p.prime) : Λ p = real.log p :=
+by rw [von_mangoldt_apply, prime.min_fac_eq hp, if_pos (nat.prime_iff.1 hp).is_prime_pow]
+
+open_locale big_operators
+
+lemma von_mangoldt_sum {n : ℕ} :
+  ∑ i in n.divisors, Λ i = real.log n :=
+begin
+  refine rec_on_prime_coprime _ _ _ n,
+  { simp },
+  { intros p k hp,
+    rw [sum_divisors_prime_pow hp, cast_pow, real.log_pow, finset.sum_range_succ', pow_zero,
+      von_mangoldt_apply_one],
+    simp [von_mangoldt_apply_pow (nat.succ_ne_zero _), von_mangoldt_apply_prime hp] },
+  intros a b ha' hb' hab ha hb,
+  simp only [von_mangoldt_apply, ←sum_filter] at ha hb ⊢,
+  rw [mul_divisors_filter_prime_pow hab, filter_union,
+    sum_union (disjoint_divisors_filter_prime_pow hab), ha, hb, nat.cast_mul,
+    real.log_mul (nat.cast_ne_zero.2 ha'.ne') (nat.cast_ne_zero.2 hb'.ne')],
+end
+
+lemma von_mangoldt_mul_zeta : Λ * ζ = log :=
```

Comment:
```suggestion
@[simp] lemma von_mangoldt_mul_zeta : Λ * ζ = log :=
```

Reply:
Do you also want the simp-lemma for `ζ * Λ`?

Reply:
Seems helpful, I'll add!